frappe/utils/caching.py

```diff
@@ -128,3 +128,54 @@ def site_cache_wrapper(*args, **kwargs):
 		return time_cache_wrapper(ttl)
 
 	return time_cache_wrapper
+
+
+def redis_cache(ttl: int | None = None, maxsize: int | None = None) -> Callable:
+	"""Decorator to cache method calls and its return values in Redis"""
+
+	def time_cache_wrapper(func: Callable = None) -> Callable:
+		func_key = f"{func.__module__}.{func.__name__}"
+
+		def clear_cache():
+			"""Clear cache for this function for all sites if not specified."""
+			frappe.cache().delete_key(func_key)
+
+		func.clear_cache = clear_cache
+
+		if ttl is not None and not callable(ttl):
+			func.ttl = ttl
+			func.expiration = datetime.utcnow() + timedelta(seconds=func.ttl)
+
+		if maxsize is not None and not callable(maxsize):
+			func.maxsize = maxsize
+
+		@wraps(func)
+		def redis_cache_wrapper(*args, **kwargs):
+			func_call_key = __generate_request_cache_key(args, kwargs)
+
+			if hasattr(func, "ttl") and datetime.utcnow() >= func.expiration:
+				func.clear_cache()
```

Comment:
This feels funky, we clear ALL cache if TTL is provided? Shouldn't TTL just be for a key?